rust/cache/src/disk/storage.rs

```diff
@@ -230,16 +230,21 @@ fn to_cache_value(entry: DirEntry) -> Result<CacheValue, String> {
     let key = parse_filename(filename.as_str())
         .ok_or_else(|| format!("{filename} doesn't follow naming convention"))?;
 
-    let header = verify_header(entry).map_err(|e| format!("{filename} invalid header: {e:?}"))?;
+    let header = if let Some(h) = Header::attempt_from_key(key) {
+        h
+    } else {
+        debug!("Warning: Parsing header from filename for {filename} failed; loading from file.");
+        verify_header(entry).map_err(|e| format!("{filename} invalid header: {e:?}"))?
+    };
 
     let file_size = metadata_size(&metadata) - header.get_header_len();
 
     Ok(CacheValue {
         size: file_size,
         version: 1,
-        key,
         block_size: header.block_size,
         block_idx: header.block_idx,
+        key: header.key,
```

Comment:
If `Header::attempt_from_key(key)` somehow fails and header is set from `verify_header`, then `header.key` is only the `request.metadata().name()` instead of the entire `"{}.{}.{}",
        request.metadata().name(),
        request.block_range().idx(),
        request.block_size(),`

Reply:
From what I can trace, the name is the format!(...) thing.  This took a while to realize.